core/network/endpoint_bindings_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package network_test
+
+import (
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/network"
+)
+
+var _ = gc.Suite(&bindingsMockSuite{})
+
+type bindingsMockSuite struct {
+	testing.IsolationSuite
+}
+
+func (s *bindingsMockSuite) TestMapBindingsWithSpaceNames(c *gc.C) {
+	infos := s.expectedSpaceInfos()
+
+	initial := map[string]string{
+		"db":      "2",
+		"testing": "3",
+		"empty":   network.AlphaSpaceId,
+	}
+
+	withSpaceNames, err := network.MapBindingsWithSpaceNames(initial, infos)
+	c.Assert(err, jc.ErrorIsNil)
```

Comment:
Consider changing these `Assert` to `Check` for all of these tests. No reason to stop the tests if they can just keep running. Just a nit and if you don't agree keep going :)